lib/src/source/hosted.dart

```diff
@@ -298,15 +298,19 @@ class BoundHostedSource extends CachedSource {
     var cacheDir = p.join(systemCacheRoot, _urlToDirectory(source.defaultUrl));
     if (!dirExists(cacheDir)) return [];
 
-    return listDir(cacheDir).map((entry) {
-      try {
-        return Package.load(null, entry, systemCache.sources);
-      } catch (error, stackTrace) {
-        log.fine("Failed to load package from $entry:\n"
-            "$error\n"
-            "${Chain.forTrace(stackTrace)}");
-      }
-    }).toList();
+    return listDir(cacheDir)
+        .map((entry) {
+          try {
+            return Package.load(null, entry, systemCache.sources);
+          } catch (error, stackTrace) {
+            log.fine("Failed to load package from $entry:\n"
+                "$error\n"
+                "${Chain.forTrace(stackTrace)}");
+            return null;
+          }
+        })
+        .where((e) => e != null)
```

Comment:
I think this fixed a real bug, in cases where loading package from the file-system fails.

This is only called in `pub cache list`, so I don't think it's very important, but still nice :)

Reply:
Yes filtering out those nulls seems to be the right thing

Reply:
It would be better to split behavior changes and hint fixes in separate commits.